### PR TITLE
Bps 765 Add S2S config

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-765

### Change description ###
Added S2S configuration for payment processor.
A new microservice key is added for the payment processor in https://github.com/hmcts/service-auth-provider-app/pull/196

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/bulk-scan-payment-processor/values.yaml [hunk 1]
```diff
@@ -4,11 +4,14 @@ java:
...
   environment:
     PAYMENTS_QUEUE_MAX_DELIVERY_COUNT: "5"
+    S2S_NAME: "bulk_scan_payment_processor"
+    S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
   image: 'https://hmctspublic.azurecr.io/bulk-scan/payment-processor:latest'
   keyVaults:
     "bulk-scan":
       resourceGroup: bulk-scan
       secrets:
+        - microservicekey-bulk-scan-payment-processor
```
Comment (doncem) [inline charts/bulk-scan-payment-processor/values.yaml]: perhaps `s2s-secret-payment-processor`?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+}
+
+# Copy s2s key from shared to local vault
+data "azurerm_key_vault" "local_key_vault" {
+  name = "${local.vaultName}"
+  resource_group_name = "${local.vaultName}"
+}
+
+resource "azurerm_key_vault_secret" "local_s2s_secret" {
+  name         = "microservicekey-bulk-scan-payment-processor"
```
Comment (doncem) [inline infrastructure/main.tf]: same as comment before

File: src/main/resources/bootstrap.yaml [hunk 3]
```diff
@@ -9,3 +9,4 @@ spring:
         site-id-divorce: SITE_ID_DIVORCE
         site-id-finrem: SITE_ID_FINREM
         payments-queue-listen-connection-string: PAYMENTS_QUEUE_READ_CONNECTION_STRING
+        microservicekey-bulk-scan-payment-processor: S2S_SECRET
```
Comment (doncem) [inline src/main/resources/bootstrap.yaml]: and here too then

File: charts/bulk-scan-payment-processor/values.yaml (hunk 1 above)
Comment (lgonczar) [reply charts/bulk-scan-payment-processor/values.yaml]: `microservicekey` prefix is needed for some automation in terraform for S2S. On our side we can call the secret sth like `s2s_secret` or something

File: charts/bulk-scan-payment-processor/values.yaml [hunk 4]
```diff
@@ -4,11 +4,14 @@ java:
...
   environment:
     PAYMENTS_QUEUE_MAX_DELIVERY_COUNT: "5"
+    S2S_NAME: "bulk_scan_payment_processor"
+    S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
   image: 'https://hmctspublic.azurecr.io/bulk-scan/payment-processor:latest'
   keyVaults:
     "bulk-scan":
       resourceGroup: bulk-scan
       secrets:
+        - s2s-secret-payment-processor
```
Comment (lgonczar) [inline charts/bulk-scan-payment-processor/values.yaml]: How about just `s2s-secret`?

Comment (lgonczar) [reply charts/bulk-scan-payment-processor/values.yaml]: That long name was needed on S2S terraform side, as it deals with all service's secrets. To us it's just a secret for s2s.

Comment (doncem) [reply charts/bulk-scan-payment-processor/values.yaml]: it's duplicated by s2s-secret which is already in bulk scan vault

Comment (lgonczar) [reply charts/bulk-scan-payment-processor/values.yaml]: aaa, ok

**Comment by aliveni** [comment]
> ...and missing version bump in charts

lost the change. updated now

**Pull request merged**